Resources/config/imagine.xml

```diff
@@ -139,14 +141,18 @@
             <argument type="service" id="logger" on-invalid="ignore" />
         </service>
 
-        <!-- Controller -->
+        <service id="Liip\ImagineBundle\Service\FilterService" alias="liip_imagine.service.filter" />
 
-        <service id="Liip\ImagineBundle\Controller\ImagineController" public="true">
-            <argument type="service" id="liip_imagine.service.filter" />
-            <argument type="service" id="liip_imagine.data.manager" />
-            <argument type="service" id="liip_imagine.cache.signer" />
+        <!-- Config -->
+
+        <service id="Liip\ImagineBundle\Config\Controller\ControllerConfig" public="false">
+            <argument><!-- will be injected by FileSystemLoaderFactory --></argument>
         </service>
 
+        <!-- Controller -->
+
+        <service id="Liip\ImagineBundle\Controller\ImagineController" public="true" autowire="true" />
```

Comment:
afaik best practice for bundles is to not use autowire as it can be a bit fragile.

Reply:
yes, you're wright